### PR TITLE
perf: Optimize subagent initialization with lazy loading and flexible config

### PR DESCRIPTION
**Pull request by wuman001**
## Summary

This PR optimizes subagent initialization through lazy loading, removes hardcoded API URL defaults, and fixes a critical bug in system prompt updates.

### 1. Lazy Initialization ⚡
- **Problem**: `sync_exec` in `Agent.__init__` caused nested event loop conflicts
- **Solution**: Defer agent.md scanning until first `spawn()` call (async context)
- **Benefit**: Improved startup performance for multi-agent applications

### 2. Flexible API Configuration 🔧
- **Problem**: Hardcoded `https://api.openai.com/v1` in multiple config classes
- **Solution**: Change `llm_base_url` defaults to `None`, fallback to env vars
- **Benefit**: Reduces coupling, improves multi-provider flexibility

### 3. 🐛 Critical Bug Fix: System Prompt Preservation
- **Problem**: Updating "Available Subagents" section **destroyed all content after it**
  - Previous logic: `split("## Available Subagents")` → only kept `parts[0]`
  - User-defined rules, guidelines, and instructions were **permanently lost**
- **Solution**: Non-destructive update that preserves all sections
  - Find next section marker (`## `) after "Available Subagents"
  - Reconstruct: `before + new_subagents + after`
- **Impact**: Prevents loss of critical user-configured instructions

## Implementation Details

**Core Changes:**

**Lazy Loading:**
- `aworld/agents/llm_agent.py`: Replace sync scanning with lazy init
- `aworld/core/agent/subagent_manager.py`: Add `_ensure_agent_md_scanned()`
- `aworld/core/tool/builtin/spawn_subagent_tool.py`: Trigger lazy scan on spawn

**Flexible Config:**
- `aworld/config/conf.py`: `llm_base_url` defaults to `None`
- `aworld/core/memory.py`: Remove URL default
- `aworld/core/context/amni/retrieval/embeddings/base.py`: Remove URL default

**Bug Fix:**
- `aworld/agents/llm_agent.py:355-375`: Enhanced system prompt update logic
- `tests/core/agent/test_system_prompt_update.py`: Comprehensive preservation tests

**Test Coverage:**
- ✅ Lazy initialization: 6/6 passed
- ✅ Edge cases: 4/4 passed
- ✅ System prompt preservation: 3/3 passed
- ✅ Core unit tests: 32/38 passed (6 E2E tests pending async mock fixes)

## Bug Fix Example

**Before (Destructive):**
```python
# System prompt with multiple sections
"## Available Subagents\n...\n## Important Rules\nBe polite\n## Context\nUse memory"

# After update → Rules and Context lost!
"## Available Subagents\n..."  # ❌
```

**After (Preserved):**
```python
# System prompt with multiple sections
"## Available Subagents\n...\n## Important Rules\nBe polite\n## Context\nUse memory"

# After update → All sections preserved!
"## Available Subagents\n...\n## Important Rules\nBe polite\n## Context\nUse memory"  # ✅
```

## Compatibility

✅ **No Breaking Changes**
- Backward compatible with existing code
- Maintains functional equivalence
- OpenAI provider has fallback: `os.getenv("OPENAI_ENDPOINT", "https://api.openai.com/v1")`

## Testing

```bash
# Lazy initialization tests
python -m pytest tests/core/agent/test_lazy_initialization.py -v

# Edge case tests
python -m pytest tests/core/agent/test_subagent_cloning_edge_cases.py -v

# System prompt preservation tests
python -m pytest tests/core/agent/test_system_prompt_update.py -v

# All core agent tests (excluding E2E)
python -m pytest tests/core/agent/ -v --ignore=tests/core/agent/test_lazy_integration_e2e.py -k "not spawn"
```

## Commits

1. **feat**: Add lightweight subagent orchestration mechanism
2. **perf**: Optimize subagent initialization with lazy loading and flexible config
3. **fix**: Preserve system prompt content when updating subagents section
4. **chore**: Add temp file patterns to .gitignore

## Review Checklist

- [x] Remove `sync_exec` from agent initialization
- [x] Implement lazy loading mechanism
- [x] Remove hardcoded API URLs
- [x] Fix destructive system prompt update
- [x] Add comprehensive test coverage (3 new test files)
- [x] Update .gitignore for temp files
- [x] Maintain backward compatibility

### CODE REVIEW

**Comment by wuman001** [comment]
## 🎯 Bug Fix Update

Thanks for the detailed review! I've addressed the **critical destructive system prompt update** issue you identified:

### ✅ Fixed: System Prompt Preservation

**Problem (as you noted):**
> The logic for updating the system prompt is destructive. By splitting on the section header and only keeping parts[0], any content that previously existed after the "## Available Subagents" section will be permanently lost.

**Solution Implemented:**
- Enhanced update logic to find the next section marker (`## `)
- Preserve all content after "Available Subagents"
- Non-destructive reconstruction: `before + new_section + after`

**Commit:** fc72c6d7
**Test:** tests/core/agent/test_system_prompt_update.py (3/3 tests passed)

**Example:**
```python
# Before fix
"## Available Subagents\n...\n## Rules\nBe polite"
→ Update → "## Available Subagents\n..." (Rules lost! ❌)

# After fix
"## Available Subagents\n...\n## Rules\nBe polite"
→ Update → "## Available Subagents\n...\n## Rules\nBe polite" (Preserved! ✅)
```

### 📊 Current Status

**Test Coverage:**
- ✅ 32/38 core tests passing
- ✅ System prompt preservation: 3/3 passed
- ✅ Lazy initialization: 6/6 passed
- ✅ Edge cases: 4/4 passed

The PR is now ready for re-review. Please let me know if you need any clarifications on the other points!

File: aworld/agents/llm_agent.py [hunk 1]
```diff
@@ -260,6 +340,36 @@ def desc_transform(self, context: Context) -> None:
...
+                    await self.subagent_manager.register_team_members(swarm)
+
+                    # Update system prompt with newly registered subagents
+                    subagent_section = self.subagent_manager.generate_system_prompt_section()
+                    if subagent_section:
+                        # Replace old subagent section or append
+                        if "## Available Subagents" in self.system_prompt:
+                            # Update existing section (split and replace)
+                            parts = self.system_prompt.split("## Available Subagents")
+                            self.system_prompt = parts[0].rstrip() + "\n\n" + subagent_section
```
[the remark this answers is not in the document]
Comment (wuman001) [reply aworld/agents/llm_agent.py]: resolve

**Review by ZhuangCY — APPROVED** [review]
